docs/DeveloperGuide.md

```diff
@@ -1,34 +1,330 @@
+
+#### <a id="event"><ins>2.6 Timetable</ins></a>
+
+The timetable component stores an array of events. On launch, an empty timetable will be created. All stored events will be loaded via the StorageManger. 
+
+Timetable handles adding, deleting and getting all instances of stored events in a given time period.
+
+#### <a id="tag"><ins>2.7 Tags</ins></a>
+
+The class diagram below denotes the relationship between the TagManager and the Taggable Objects (Notes and Events).
+
+<p align="center">
+   <img alt="TaggableObject" src="diagrams/out/TaggableObject.png"/>
+   <br><em>Figure 10</em>
+</p>
+ 
+💡 As the focus of this diagram is on Tag, TaggableObject and TagManager, the variables and methods of Notes and Events are omitted.
+ 
+Notes and Events inherit from the abstract class, TaggableObject, and TagManager contains a map of individual unique tags to an ArrayList of TaggableObjects. The TagManager handles the creation, deletion as well as the tagging and untagging of tags from notes or events.
+
+#### <a id="storage"><ins>2.8 Storage</ins></a>
+The StorageManager saves and loads data to text files. On launch, the storage manager checks for existing directories that may contain previously saved data, otherwise it creates the necessary directories. Following that, it will load the previously saved notes and events from the text files into NotUS.Below is the class diagram representing the relationship between the StorageManager, Timetable, Notebook, TagManager and ParserManager.
+
+<p align="center">
+   <img alt="StorageManagerClassDiagram" src="diagrams/out/StorageManager.png"/>
+   <br><em>Figure 11</em>
+</p>
+
+While loading information is passed to the parser manager to prepare the information to be added. Following that, the respective Add Command will be called to add the event/note to the program Below is the sequence for loading the notes and events when the program first starts up. 
+
+<p align="center">
+   <img alt="StorageManagerObjectDiagram" src="diagrams/out/StorageManagerObject.png"/>
+   <br><em>Figure 12</em>
+</p>
+
+#### <a id="ui"><ins>2.9 User Interface</ins></a>
+
+The Formatter class handles the formatting of the Note(s), Event(s) and message(s) which is then displayed to the user. Any changes to the layout or information to display will be done in this class. This class only contains static methods to eliminate the need of a Formatter object.
+
+<p align="center">
+   <img alt="Formatter" src="diagrams/out/Formatter.png"/>
+   <br><em>Figure 13</em>
+</p>
+
+#### <a id="exception"><ins>2.10 System Exception</ins></a>
+
+The System Exception Enumeration contains all the possible types of exception with specific messages.
+
+<p align="center">
+   <img alt="SystemExceptionEnum1" src="diagrams/out/SystemExceptionEnum1.png"/>
+   <img alt="SystemExceptionEnum2" src="diagrams/out/SystemExceptionEnum2.png"/>
+   <br><em>Figure 14</em>
+</p>
+
```

Comment:
Class diagrams used for almost everything makes it easier to use the developer guide. It is all very clearly described and mentioned. Good job guys